### ISSUE TITLE
 "Critical-Origin-Trial: Tpcd" is not behaving according to documentation 

### ISSUE THREAD

**Issue by pdisessa-de** [GoogleChromeLabs/privacy-sandbox-dev-support#275]
I am attempting to use the header "Critical-Origin-Trial: Tpcd" as described in the 3pcd documentation here https://developers.google.com/privacy-sandbox/3pcd/temporary-exceptions/third-party-deprecation-trial

We are loading our embedded 3rd party content into an iframe inside the Canvas Learning Management System.  The user clicks on a link in Canvas' navigation which creates a POST request to our system to load the content.  That first request immediately sets our 3rd party cookies.  My expectation is that "Critical-Origin-Trial: Tpdc" would cause the first request to be reloaded after the user clicks the link if the TPCD trial feature was not initialized.  However that does not occur, and our cookies are blocked.  Upon clicking the link a second time, the content does load though.  I confirmed in Chrome's Developer tool's Application tab that the token was loaded both on the first and second clicks. This tells me that the "Origin-trial" header is working and the token is valid, but the "Critical-Origin-Trial" is not working as documented.

My theory is since Canvas is using a POST request to our system in order to load the embedded content into the iframe, that Critical-Origin-Trial doesn't properly reload it.  

Steps:
1. Set all three Chrome settings as described in documentation linked above.
chrome://flags#third-party-cookie-deprecation-trial -> enabled
chrome://flags/#tracking-protection-3pcd -> enabled
chrome://flags/#tpcd-metadata-grants -> disabled

2. Clear browser cookies and cache

3. Login to our Canvas test user (unable to provide login cred or url at this time)

4. Open Chrome Dev tools

5. Click on link which loads our embedded content
    - Observed that content does not load and the request does not get reloaded in Network tab
    - Observed that the 3rd party cookies are blocked
    - Ensured both Critical-Origin-Trial and Origin-Trial are set correctly in the response headers of the request in network tab

6. Click the link again
   - Observed that the content does load properly

I've attached screenshots showing the Network tab on the first click and second click, as well as one showing the Application tab.

I'm using Chrome Version 122.0.6261.95 (Official Build) (64-bit) and Third-Party Cookie Deprecation Metadata - Version: 2024.3.5.1

![image-20240229-162240](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/63816475/237ecbfa-15b2-47f3-8795-8904bd8daa7a)

![Screenshot 2024-03-06 144102](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/63816475/d31e37d0-e8b7-4dc6-af05-d4e52456ed3e)

**Comment by miketaylr** [comment]
Mind taking a look @hoodjoshua?

**Comment by hoodjoshua** [comment]
> My theory is since Canvas is using a POST request to our system in order to load the embedded content into the iframe, that Critical-Origin-Trial doesn't properly reload it.

You are correct. The Critical-Origin-Trial header is only effective on navigation requests made using "safe" HTTP request methods (as defined by [section 4.2.1 of RFC 7231](https://www.rfc-editor.org/rfc/rfc7231#section-4.2.1)). So, if possible, loading the iframe using a GET should work.
